src/snowflake/snowpark/mock/_stage_registry.py

```diff
@@ -413,6 +448,104 @@ def read_file(
                 result_df = pd.concat([result_df, df], ignore_index=True)
             result_df.sf_types = result_df_sf_types
             return result_df
+        elif file_format == "json":
+            infer_schema_opt = options.get("INFER_SCHEMA", False)
+
+            result_df = TableEmulator()
+            result_df_sf_types = {}
+
+            if not infer_schema_opt:
+                # if infer schema option is False, then snowflake converts the data into
+                # a single column table, values are treated as raw strings
+                assert len(schema) == 1, (
+                    f"[Local Testing] Unexpected schema length {len(schema)} when loading "
+                    f"json data without inferring schema."
+                )
+                column_name = analyzer.analyze(schema[0])
+                column_series = ColumnEmulator(
+                    data=None,
+                    dtype=object,
+                    name=column_name,
+                    sf_type=ColumnType(schema[0].datatype, schema[0].nullable),
```

Comment:
Shouldn't the datatype always be `Variant` in this case? Does specifying other data types work?

Reply:
I tested and you are right! changed to use variant type and set nullable to True (this is my test observation that snowflake always set it to True)